yaksh/templatetags/custom_filters.py

```diff
@@ -91,7 +91,24 @@ def replace_spaces(name):
 
 @register.simple_tag
 def get_questions_by_type(all_questions, question_type):
-    return [question for question in all_questions if question.type == question_type]
+    return [question for question in all_questions
+            if question.type == question_type]
+
+
+@register.simple_tag
+def get_question_by_index(all_questions, questions_by_type):
+    return [index for index, item in enumerate(all_questions)
+            if item in set(questions_by_type)]
+
+
+@register.simple_tag
+def get_categories(question_types):
+    categories = {}
+    objectives = [type_ for type_ in question_types if type_ != 'code']
+    programming = [type_ for type_ in question_types if type_ == 'code']
```

Comment:
```
categories = {}
    categories["objectives"] = []
    categories["programming"] = []
    for type_ in all_question_types:
        if type_ != 'code':
            categories["objectives"].append(type_)
        else:
            categories["programming"].append(type_)
```
Using this in views.py instead of custom_filter.

Reply:
No need of this loop instead use the filter to get questions categories.

Reply:
I agree with @adityacp 
I also feel that the code is too complex, can be simplified.
Also, the assignment upload question is mainly non-objective type, so need to check that as well!
AnswerPaper stores the order of the questions. You can also think of having a class method that will simply give you all the objectives and the code questions for a given answerpaper.
So in the template you don't need to process anything complex or even the filter!

Reply:
@adityacp i came up with something like this 
```
    questions = paper.question_paper.fixed_questions.all()
    objectives = questions.all().exclude(type='code').exclude(type='upload')
    upload = questions.filter(type='upload')
    programming = questions.filter(type='code')
```